pkg/jobs/registry.go

```diff
@@ -141,8 +138,19 @@ type Registry struct {
 		// jobs scheduled inside a transaction, they will show in this map but will
 		// only be run when the transaction commits.
 		adoptedJobs map[jobspb.JobID]*adoptedJob
+
+		// waiting is a set of jobs for which we're waiting to complete. In general,
+		// we expect these jobs to have been started with a claim by this instance.
+		// That may not have lasted to completion. Separately a goroutine will be
+		// passively polling for these jobs to complete. If they complete locally,
+		// the waitingSet will be updated appropriately.
+		waiting map[jobspb.JobID]map[*waitingSet]struct{}
```

Comment:
nothing wrong w/ this..  but I wonder if we need this map of map of pointer complexity?
Wouldn't just a slice do? I understand we have to iterate or whatnot, but we're not expecting thousands of entries in there, do we?